src/coreclr/jit/llvm.cpp

```diff
@@ -755,6 +756,19 @@ Value* Llvm::genTreeAsLlvmType(GenTree* tree, Type* type)
         }
         return _builder.getInt({(unsigned int)type->getPrimitiveSizeInBits().getFixedSize(), (uint64_t)tree->AsIntCon()->IconValue(), true});
     }
+    //// from i1 (e.g. from GT_EQ) to wider int
+    if (v->getType()->isIntegerTy() && type->isIntegerTy() &&
+        v->getType()->getPrimitiveSizeInBits().getFixedSize() < type->getPrimitiveSizeInBits().getFixedSize())
+    {
+        return _builder.CreateZExt(v, type);
+    }
```

Comment:
Still needed? It seems like `genTreeAsLlvmType` is similar to `consumeValue`, but different. Put the logic for bitcasting pointers to integers and back into `consumeValue` as well?